packages/flutter_test/lib/src/controller.dart

```diff
@@ -696,8 +697,80 @@ class LiveWidgetController extends WidgetController {
 
   @override
   Future<List<Duration>> handlePointerEventRecord(List<PointerEventRecord> records) {
-    // TODO(CareF): This will be implemented after we decide what should be the
-    // correct pumping strategy.
-    throw UnimplementedError;
+    assert(records != null);
+    assert(records.isNotEmpty);
+    return TestAsyncUtils.guard<List<Duration>>(() async {
+      // hitTestHistory is an equivalence of _hitTests in [GestureBinding]
+      final Map<int, HitTestResult> hitTestHistory = <int, HitTestResult>{};
+      final List<Duration> handleTimeStampDiff = <Duration>[];
+      DateTime startTime;
+      for (final PointerEventRecord record in records) {
+        final DateTime now = clock.now();
+        startTime ??= now;
+        // So that the first event is promised to receive a zero timeDiff
+        final Duration timeDiff = record.timeDelay - now.difference(startTime);
+        if (timeDiff.isNegative) {
+          // Flush all past events
+          handleTimeStampDiff.add(timeDiff);
+          for (final PointerEvent event in record.events) {
+            _handlePointerEvent(event, hitTestHistory);
+          }
+        } else {
+          await Future<void>.delayed(timeDiff);
+          handleTimeStampDiff.add(
+            record.timeDelay - clock.now().difference(startTime),
```

Comment:
Isn't it the same as `timeDiff` so that it's basically the same as L714?

Reply:
No because `Future.delay` doesn't promise exact delay time.

Reply:
I think I misunderstood the question for the above reply. Ideally L714 should give a negative value but this should give zero (and is indeed zero in #60796 's unit test where it's using a fake async), but since `Future.delay` is not exact, there will be a little extra delay.

Reply:
I see. Can you add a comment saying "Recalaulate time diff because (...)"?

Reply:
Done.